payments/models.py

```diff
@@ -296,6 +298,74 @@ def name(self):
     )
 
 
+class TalpaProductAccountingManager(models.Manager):
+    def get_product_accounting_for_order(self, order):
+        """
+        Talpa ecom requirements for accounting of products defined in Talpa:
+        * Orders in the EAST/WEST regions need to be for separate products
+        * Orders for winter storage (including storage on ice) and berths need to be separate products
+        """
+
+        if order.order_type == OrderType.ADDITIONAL_PRODUCT_ORDER:
+            if (
+                order.order_lines.count() == 1
+                and order.order_lines.all().first().product.service
+                == ProductServiceType.STORAGE_ON_ICE
+            ):
+                product_type = TalpaProductType.WINTER
+                region = order.lease.berth.pier.harbor.region
+            else:
+                raise TalpaProductAccountingNotFoundError(
+                    _("Only STORAGE_ON_ICE additional product is supported")
+                )
+        else:
+            if order.lease_order_type in (
+                LeaseOrderType.WINTER_STORAGE_ORDER,
+                LeaseOrderType.UNMARKED_WINTER_STORAGE_ORDER,
+            ):
+                region = order.product.winter_storage_area.region
+                product_type = TalpaProductType.WINTER
+            elif order.lease_order_type in (
+                LeaseOrderType.BERTH_SWITCH_ORDER,
+                LeaseOrderType.RENEW_BERTH_ORDER,
+                LeaseOrderType.NEW_BERTH_ORDER,
+            ):
+                region = order.lease.berth.pier.harbor.region
+                product_type = TalpaProductType.BERTH
+            else:
+                raise TalpaProductAccountingNotFoundError(
+                    _("TalpaProductAccounting parameters for order")
+                )
+
+        if region is None:
+            raise TalpaProductAccountingNotFoundError(
+                _("Region for order could not be determined")
+            )
+        if product_type is None:
+            raise TalpaProductAccountingNotFoundError(
+                _("Talpa product type for order could not be determined")
+            )
+        return self.get(region=region, product_type=product_type)
+
+
+class TalpaProductAccounting(UUIDModel):
```

Comment:
We should add a unique constraint for `region` + `product_type` to ensure we don't have multiple results when fetching the accounting info for an order.